### ISSUE TITLE
Aliasing when using vdpau with radeonsi

### ISSUE THREAD

**Issue by dolohow** [mpv-player/mpv#3317]
### mpv version and platform

0.18 on x86_64 GNU/Linux

### Reproduction steps

Start a movie with vdpau hardware decoding and opengl[-hq] with radeonsi driver.  Observe aliasing effect on rough edges.

### Expected behavior

No aliasing effect.

### Actual behavior

Aliasing on rough edges.

### Log files

[vdpauinfo](http://sprunge.us/FIKJ)

### Other

This is regression caused by https://github.com/mpv-player/mpv/commit/7be37337f.

**Comment by wm4** [comment]
Need to find out whether upstream is going to fix this (or whether it's actually our fault), and if not, work it around.

**Comment by wm4** [comment]
Also, explanation: it somehow has to do with mapping the video surfaces as OpenGL textures, and reinterleaving them to a "progressive" texture.

**Comment by haasn** [comment]
I'd like to reiterate that I think vdpau-mixer and vdpau-raw should be different hwdec options.

**Comment by wm4** [comment]
Well the user can actually trigger the mixer code path with ``--vf=vdpaupp``. Messing this into the hwdec option wouldn't work well.

**Comment by wm4** [comment]
This is the shader we use to reinterleave the fields:
```
[vo/opengl-hq] [  1] color = fract(gl_FragCoord.y / 2) < 0.5
[vo/opengl-hq] [  2]       ? texture(texture0, texcoord0)
[vo/opengl-hq] [  3]       : texture(texture1, texcoord1);
```
In my opinion, this should be fine. We use it with NEAREST filtering (otherwise it obviously wouldn't work).

Does anyone want to report this upstream?

**Comment by dolohow** [comment]
Reported in mesa issue tracker [#96860](https://bugs.freedesktop.org/show_bug.cgi?id=96860).

**Comment by wm4** [comment]
You should probably post some screenshots too. It's hard to guess what's going wrong.

**Comment by andreaskem** [comment]
I noticed this as well using the radeon driver with vdpau. However, upgrading to mesa 12.0.0 apparently fixed it.

**Comment by dolohow** [comment]
Fixed in mesa 12.0.1
